packages/frontend/component/src/ui/date-picker/week-date-picker.tsx

```diff
@@ -0,0 +1,248 @@
+
+  useImperativeHandle(handleRef, () => ({
+    setCursor,
+  }));
+
+  const displayDays = allDays.slice(...range);
+
+  const updateRange = useCallback(
+    (newRange: [number, number]) => {
+      if (range && newRange[0] === range[0] && newRange[1] === range[1]) return;
+      setRange(newRange);
+    },
+    [range]
+  );
+
+  const onNext = useCallback(() => {
+    if (viewPortSize === 7) {
+      setCursor(cursor.add(1, 'week'));
+    } else if (range[1] === 7) {
+      setCursor(cursor.add(1, 'week'));
+      updateRange([0, viewPortSize]);
+    } else {
+      const end = Math.min(range[1] + viewPortSize, 7);
+      const start = Math.min(range[0] + viewPortSize, end - viewPortSize);
+      updateRange([start, end]);
+    }
+  }, [cursor, range, updateRange, viewPortSize]);
+  const onPrev = useCallback(() => {
+    if (viewPortSize === 7) {
+      setCursor(cursor.add(-1, 'week'));
+    } else if (range[0] === 0) {
+      setCursor(cursor.add(-1, 'week'));
+      updateRange([7 - viewPortSize, 7]);
+    } else {
+      const start = Math.max(range[0] - viewPortSize, 0);
+      const end = Math.max(range[1] - viewPortSize, start + viewPortSize);
+      updateRange([start, end]);
+    }
+  }, [cursor, range, updateRange, viewPortSize]);
+  const onDayClick = useCallback(
+    (day: dayjs.Dayjs) => {
+      onChange?.(day.format(format));
+    },
+    [onChange]
+  );
+
+  // Observe weekRef to update viewPortSize
+  useEffect(() => {
+    const el = weekRef.current;
+    if (!el) return;
+
+    const resizeObserver = new ResizeObserver(entries => {
+      const rect = entries[0].contentRect;
+      const width = rect.width;
+      if (!width) return;
+
+      const minWidth = 30;
+      const gap = 4;
+      const viewPortCount = Math.floor(width / (minWidth + gap));
+      setViewPortSize(Math.max(1, Math.min(viewPortCount, 7)));
+      setDense(width < 300);
+    });
+    resizeObserver.observe(el);
+
+    return () => {
+      resizeObserver.disconnect();
+    };
+  }, []);
+
+  // update allDays when cursor changes
+  useEffect(() => {
+    const firstDay = dayjs(cursor).startOf('week');
+    if (allDays[0] && firstDay.isSame(allDays[0], 'day')) return;
+
+    setAllDays(
+      Array.from({ length: 7 }).map((_, index) =>
+        firstDay.add(index, 'day').startOf('day')
+      )
+    );
+  }, [allDays, cursor]);
+
+  // when viewPortSize changes, reset range
+  useEffect(() => {
+    if (viewPortSize >= 7) updateRange([0, 7]);
+    else {
+      const end = Math.min(7, range[0] + viewPortSize);
+      const start = Math.max(0, end - viewPortSize);
+      updateRange([start, end]);
+    }
+  }, [range, updateRange, viewPortSize]);
+
+  // when value changes, reset cursor
+  useEffect(() => {
+    value && setCursor(dayjs(value));
+  }, [value]);
+
+  // TODO: keyboard navigation
+  useEffect(() => {
+    if (!weekRef.current) return;
+    const el = weekRef.current;
+    const onKeyDown = (e: KeyboardEvent) => {
```

Comment:
may need to stop event propagation if it is handled

Reply:
is it ok？not sure

Reply:
cc @CatsJuice

Reply:
It seems fine at the moment, but adding it might be safer.

Reply:
Added